src/components/Courses/Courses.jsx

```diff
@@ -1,34 +1,53 @@
-import { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
-
+import { useSelector, useDispatch } from "react-redux";
+import { getCoursesService } from "../../services";
 import SearchBar from "./components/SearchBar/SearchBar";
 import CourseCard from "./components/CourseCard/CourseCard";
 import Button from "../../common/Button/Button";
-
-import { mockedCoursesList } from "../../constants";
-
 import styles from "./Courses.module.css";
+import { PATH_URIS } from "../../constants";
+import { BUTTON_TEXT, LOG_MESSAGES } from "./coursesStrings";
 
 function Courses() {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const courses = useSelector((state) => state.courses);
+  const [search, setSearch] = useState("");
+
+  useEffect(() => {
+    const fetchCourses = async () => {
+      try {
+        const courses = await getCoursesService();
+        dispatch(getCoursesService(courses));
+      } catch (error) {
+        console.error(LOG_MESSAGES.FETCH_ERROR, error);
+      }
+    };
+
+    fetchCourses();
+  }, [dispatch]);
 
   const createCourseButtonHandler = () => {
-    navigate("/courses/add");
+    navigate(PATH_URIS.ADD_COURSE);
   };
 
-  const [search, setSearch] = useState("");
-
-  const filteredCourseList = mockedCoursesList.filter(
-    (course) =>
-      course.title.toLowerCase().includes(search.toLowerCase()) ||
-      course.id.toLowerCase().includes(search.toLowerCase()),
-  );
+  const filteredCourseList = Array.isArray(courses)
```

Comment:
we can omit this check if we do `const courses = useSelector((state) => state.courses) || [];` at the top (or even make sure we pass an empty array with the selector

Reply:
done